sdk/src/trace/tracer.cc

```diff
@@ -38,14 +40,19 @@ nostd::unique_ptr<trace_api::Span> Tracer::StartSpan(
       sampler_->ShouldSample(nullptr, trace_api::TraceId(), name, options.kind, attributes);
   if (sampling_result.decision == Decision::NOT_RECORD)
   {
-    return nostd::unique_ptr<trace_api::Span>{new (std::nothrow)
-                                                  trace_api::NoopSpan{this->shared_from_this()}};
+    auto span = nostd::shared_ptr<trace_api::Span>{
+        new (std::nothrow) trace_api::NoopSpan{this->shared_from_this()}};
+
+    return span;
   }
   else
   {
-    auto span = nostd::unique_ptr<trace_api::Span>{new (std::nothrow) Span{
+    auto span = nostd::shared_ptr<trace_api::Span>{new (std::nothrow) Span{
         this->shared_from_this(), processor_.load(), name, attributes, options}};
 
+    span->SetToken(new context::Token(context::RuntimeContext::Attach(
```

Comment:
This is bad. https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#i11-never-transfer-ownership-by-a-raw-pointer-t-or-reference-t ... since you are allocating and passing raw pointer.

Reply:
Changed to unique_ptr.